packages/adapter-utils/src/list_comparison.ts

```diff
@@ -274,6 +304,90 @@ export const getArrayIndexMapping = (before: Value[], after: Value[]): IndexMapp
   return orderedItems
 }
 
+const hasBeforeIndex = (change: ListChange): change is ListChange & { action: 'remove' | 'modify' } =>
+  change.action !== 'add'
+
+const hasAfterIndex = (change: ListChange): change is ListChange & { action: 'add' | 'modify' } =>
+  change.action !== 'remove'
+
+// eslint-disable-next-line consistent-return
+const toListChange = (change: DetailedChange): ListChange => {
+  // eslint-disable-next-line default-case
+  switch (change.action) {
+    case 'remove': {
+      return { action: 'remove', beforeIndex: Number(change.elemIDs?.before?.name) }
+    }
+    case 'add': {
+      return { action: 'add', afterIndex: Number(change.elemIDs?.after?.name), afterValue: change.data.after }
+    }
+    case 'modify': {
+      return {
+        action: 'modify',
+        beforeIndex: Number(change.elemIDs?.before?.name),
+        afterIndex: Number(change.elemIDs?.after?.name),
+        afterValue: change.data.after,
+      }
+    }
+  }
+}
+
+const updateList = (list: unknown[], changes: ListChange[]): void => {
+  const removalsAndModifications = changes.filter(hasBeforeIndex)
+  removalsAndModifications.forEach(({ beforeIndex }) => {
+    list[beforeIndex] = undefined
+  })
+
+  const sortedAdditionsAndModifications = _.sortBy(changes.filter(hasAfterIndex), change => change.afterIndex)
+  sortedAdditionsAndModifications.forEach(({ afterIndex, afterValue }) => {
```

Comment:
```suggestion
  _.sortBy(changes.filter(hasAfterIndex), change => change.afterIndex).forEach(({ afterIndex, afterValue }) => {
```
Also here, just personal preference.